server/pulp/plugins/rsync/publish.py

```diff
@@ -0,0 +1,522 @@
+        is_ok, output = self.call(args, include_args_in_output=True)
+        _logger.info("remote mkdir out: %s" % output)
+        shutil.rmtree(tmpdir)
+        return is_ok, output
+
+    def make_ssh_cmd(self, args=None):
+        """
+        Returns a list of arguments needed to form an ssh command for connecting to remote server.
+
+        :param args:
+        :type args: list of extra args to append to the standard ssh command
+        :return: list of arguments for ssh portion of command
+        :rtype: list
+        """
+        user = self.get_config().flatten()["remote"]['ssh_login']
+        auth_type = self.get_config().flatten()["remote"]['auth_type']
+
+        # -e 'ssh -l l_name -i key_path'
+        # use shared ssh connection for other threads
+        cmd = ['ssh', '-l', user]
+        if auth_type == "publickey":
+            key = self.get_config().flatten()["remote"]['key_path']
+            cmd += ['-i', key]
+        cmd += ['-o', 'StrictHostKeyChecking no',
+                '-o', 'UserKnownHostsFile /dev/null',
+                '-S', '/tmp/rsync_distributor-%r@%h:%p',
+                '-o', 'ControlMaster auto',
+                '-o', 'ControlPersist 10']
+        if self.get_config().flatten()["remote"]['auth_type'] == 'password':
+            password_file = os.path.join(self.get_working_dir(), str(uuid.uuid4()))
+            open(password_file, 'w').write(self.get_config().flatten()['remote']['ssh_password'])
+            cmd = ['sshpass', '-f', password_file] + cmd
+        if args:
+            cmd += args
+        return cmd
+
+    def make_authentication(self):
+        """
+        Returns a string representation of command for authenticating against a remote server.
+
+        :return: str. e.g., '-e ssh -l ssh_login_user -i /key_path'
+        :rtype: str
+        """
+        if self.get_config().flatten()["remote"]['auth_type'] == 'local':
+            return ()
+        ssh_parts = []
+        for arg in self.make_ssh_cmd():
+            if " " in arg:
+                ssh_parts.append('"%s"' % arg)
+            else:
+                ssh_parts.append(arg)
+
+        return ['-e', " ".join(ssh_parts)]
+
+    def make_full_path(self, relative_path):
+        """
+        Returns full path of a relative path on the remote server.
+
+        :param relative_path: relative path
+        :type relative_path: str
+        """
+        return os.path.join(self.get_config().flatten()["remote"]['root'], relative_path)
+
+    def make_destination(self, relative_path):
+        """
+        Parse from self.config information to make up a hostname and remote path used for rsync
+        command.
+
+        :return: str of the combination of user, host, and dir.
+        :rtype:  str
+        """
+        if self.get_config().flatten()["remote"]['auth_type'] == 'local':
+            return self.get_config().flatten()["remote"]['root']
+        relative_path = relative_path.lstrip("/")
```

Comment:
If it has a leading ``/``, it's not a relative path. Hopefully that would fail validation above, but I'd also be inclined to raise a ``ValueError`` here instead of silently "fixing" it.